Docs: document world docs and tests (#1463)
ArchipelagoMW/Archipelago: docs/world api.md

    generate_mod(src, out_file, data)
```

### Documentation

> Every World implemented should have a tutorial and a game info page. These are both rendered on the website by reading

Reviewer: Maybe?
```suggestion
Each world implementation should have a tutorial and a game info page. These are both rendered on the website by reading
```

the `.md` files in your world's `/docs` directory.

#### Game Info
The game info page is for a short breakdown of what your game is and how it works in Archipelago. Any additional

> information that may be useful for the player to know when learning your randomizer should also go here. The file format
> is `<language key>_<game name>.md`. While you can write these docs for multiple languages and can tell the `WebWorld`
> you have multiple of these docs, they currently do not render on the website.

Reviewer: I'm not a native speaker, but the "to know" seems redundant to me.
The other changes: i don't have the info page in my webworld?
```suggestion
information that may be useful to the player when learning your randomizer should also go here. The file name format
is `<language key>_<game name>.md`. While you can write these docs for multiple languages, currently only the english
version is displayed on the website.
```

#### Tutorials
Your game can have as many tutorials in as many languages as you like, with each one having a relevant `Tutorial`
defined in the `WebWorld`. The file name you use aren't particularly important, but it should be descriptive of what

> the tutorial is covering, and the name of the file must match the url extension provided in the `Tutorial`.

Reviewer: i believe `_` is translated to a `/`, and there can only be one (or exactly one)? If this is true, then that should be noted here.

### Tests

> Ever World is expected to write unit tests that cover its logic, to ensure no logic bug regressions. This can be done

Reviewer: I believe "each" is more correct. Also we use lowercase "world" in other places.
```suggestion
Each world is expected to include unit tests that cover its logic, to ensure no logic bug regressions. This can be done
```

Each world is expected to include unit tests that cover its logic, to ensure no logic bug regressions. This can be done

> by creating a `/test` package within your world package. The `__init__.py` within this folder is where the World's

Reviewer: w
```suggestion
by creating a `/test` package within your world package. The `__init__.py` within this folder is where the world's
```

TestBase, and can then define options to test in the class body, and run tests in each test method.

Example `__init__.py`
```python
from test.TestBase import WorldTestBase

class MyGameTestBase(WorldTestBase):
  game = "My Game"
```

Next using the rules defined in the above `set_rules` we can test that the chests have the correct access rules.

Example `testChestAccess.py`
```python
from . import MyGameTestBase

class TestChestAccess(MyGameTestBase):
    def testSwordChests(self):
        """Test locations that require a sword"""

>         # this builds a list of Chest2 - Chest5
>         locations = [f"Chest{i}" for i in range(2, 6)]

Reviewer: Show that it can also be a hardcoded list
```suggestion
        locations = ["Chest1", "Chest2"]
```

        items = [["Sword"]]

>         # this will test that each location can't be accessed without the items in each list in `items`, but can be
>         # once we have those items.

Reviewer: ```suggestion
        # this will test that each location can't be accessed without the "Sword", but can be accessed once obtained
```

        self.assertAccessDependency(locations, items)
    
    def testAnyWeaponChests(self):

>         """Test locations that require any weapons"""

Reviewer: ```suggestion
        """Test locations that require any weapon"""
```

        locations = [f"Chest{i}" for i in range(3, 6)]
        items = [["Sword"], ["Axe"], ["Spear"]]

>         # this will test that any weapon from 3-5 can't be accessed without these items, but can be with all of them.

Reviewer: ```suggestion
        # this will test that chests 3-5 can't be accessed without any weapon, but can be with just one of them.
```
